controllers/category.js

```diff
@@ -0,0 +1,244 @@
+    req.files.forEach(element => {
+      const file = {
+        imageName: element.originalname,
+        imagePath: element.path,
+        imageType: element.mimetype,
+        fileSize: fileSizeFormatter(element.size, 2)
+      }
+      imagesArray.push(file);
+    });
+    const categoryData = new Category({
+      categoryName: req.body.categoryName,
+      subCategories: req.body.subCategories,
+      categoryImages: imagesArray,
+      price: req.body.price,
+    });
+    await categoryData.save();
+    res.status(201).send({
+      status:true,
+      massage:'Category saved and images Uploaded Successfully',
+      data:categoryData,
+    });
+  } catch (error) {
+    res.status(400).send(error.message);
+  }
+}
+
+const fileSizeFormatter = (bytes, decimal) => {
+  if (bytes === 0) {
+    return '0 Bytes';
+  }
+  const dm = decimal || 2;
+  const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'YB', 'ZB'];
+  const index = Math.floor(Math.log(bytes) / Math.log(1000));
+  return parseFloat((bytes / Math.pow(1000, index)).toFixed(dm)) + ' ' + sizes[index];
+
+}
+
+/**
+ * @api {put} /Category/ Modify Category information
+ * @apiName PutCategory
+ * @apiGroup Category
+ *
+ * @apiParam {Number} id  Category unique ID.
+ * @apiParam {String} [categoryName] categoryName of the product.
+ * @apiParam {String} [subategories]  subCategories of the product.
+ *
+ * @apiParam {Object} [categoryImages]  CategoryImages of the product.
+ *
+ * @apiSuccessExample Success-Response:
+ *     HTTP/1.1 200 OK
+ *
+ * @apiUse CategoryNotFoundError
+ */
+
+//update data
+exports.update = async (req, res) => {
+  try {
+    
+    const updatedCartegory = await Category.findByIdAndUpdate(
+      req.params.id,
+      {
+        $set: req.body,
+      },
+      { new: true }
+    );
+    res.status(200).json(updatedCartegory);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+};
+
+
+/**
+ * @api {get} /Category/:id Get User information .
+ * @apiVersion 0.2.0
+ * @apiName GetCategory
+ * @apiGroup Category
+ *
+ * @apiParam {Number} id Category unique ID.
+ *
+ * @apiSuccess {String} categoryName  CategoryName of the Product.
+ * @apiSuccess {String} sunCategories   subCategories of the Product.
+ * 
+ * @apiSuccessExample Success-Response:
+ *     HTTP/1.1 200 OK
+ *     {
+ *       "CategoryName": "women Ethics",
+ *       "subCategories": " sarees"
+ *     }
+ *
+ * @apiError CategoryNotFound The id of the User was not found.
+ *
+ * @apiErrorExample Error-Response:
+ *     HTTP/1.1 404 Not Found
+ *     {
+ *       "error": "CategoryNotFound"
+ *     }
+ */
+
+
+//find one
+exports.getOne = async (req, res) => {
+  try {
+    const categoryData = await Category.findById(req.params.id)
+    .select('categoryName subCategories categoryImages price ')
+    res.status(200).json(categoryData)
+
+  } catch (err) {
+    res.status(500).json(err);
+  }
+};
+//countAllData
+exports.countAllCategories = async (req, res) => {

```

Comment:
we don't need different api for fetching total categories...include this in getall